src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+                context.Location());
+
+            foreach (var annotation in members.OfType<CsdlAnnotation>())
+            {
+                entityType.AddAnnotation(annotation);
+            }
+
+            return entityType;
+        }
+
+        private static CsdlElement TryParseStructuredTypeElement(string name, JsonElement element, JsonParserContext context)
+        {
+            // It also MAY contain members representing structural properties and navigation properties as well as annotations.
+            Debug.Assert(!string.IsNullOrEmpty(name), "json reader should verify that the property name string is not null or empty.");
+            Debug.Assert(context != null);
+
+            if (AnnotationJsonParser.TryParseCsdlAnnotation(name, element, context, out CsdlAnnotation annotation))
+            {
+                return annotation;
+            }
+
+            // The remaining is property or navigation property, both them are required an object
+            if (element.ValueKind != JsonValueKind.Object)
+            {
+                return null;
+            }
+
+            // so it's complex property, Property is an object
+            string kind = GetKind(element, context);
+            if (kind == "NavigationProperty")
+            {
+                return ParseCsdlNavigationProperty(name, element, context);
+            }
+
+            // The property object MAY contain the member $Kind with a string value of Property.
+            // This member SHOULD be omitted to reduce document size.
+            return ParseCsdlProperty(name, element, context);
+        }
+
+        /// <summary>
+        /// Parse the input <see cref="JsonElement"/> to the <see cref="CsdlComplexType"/>.
+        /// </summary>
+        /// <param name="name">the unqualified name of the complex type.</param>
+        /// <param name="complexObject">The json value to parse from.</param>
+        /// <param name="jsonPath">The input json value path.</param>
+        /// <returns>null or the parsed <see cref="CsdlComplexType"/>.</returns>
+        internal static CsdlComplexType ParseCsdlComplexType(string name, JsonElement complexObject, JsonParserContext context)
+        {
+            // A complex type is represented as a member of the schema object whose name is the unqualified name of the complex type and whose value is an object.
+            Debug.Assert(complexObject.ValueKind == JsonValueKind.Object);
+
+            // The complex type object MUST contain the member $Kind with a string value of ComplexType.
+            ValidateKind(complexObject, context, "ComplexType", required: true);
+
+            // It MAY contain the members $BaseType, $Abstract, and $OpenType.
+            // It also MAY contain members representing structural properties and navigation properties as well as annotations.
+            string baseType = null;
+            bool? isOpen = null;
+            bool? isAbstract = null;

```

Comment:
why not default isOpen and isAbstract to false, then only set to true if so specified?